### PR TITLE
add verbose option to initializer

### PR DESCRIPTION
**Pull request by zhreshold**
* Add verbose option to initializer, according to #5776

### CODE REVIEW

File: python/mxnet/initializer.py [hunk 1]
```diff
@@ -36,6 +38,32 @@ class Initializer(object):
     def __init__(self, **kwargs):
         self._kwargs = kwargs
 
+    def verbose(self, verbose=False, stat_func=None):
```
Comment (piiswrong) [inline python/mxnet/initializer.py]: set_verbosity

**Comment by zhreshold** [comment]
fixed and rebased

File: python/mxnet/initializer.py [hunk 2]
```diff
@@ -35,6 +37,35 @@ class Initializer(object):
     """The base class of an initializer."""
     def __init__(self, **kwargs):
         self._kwargs = kwargs
+        self._verbose = False
+        self._stat_func = None
+        self._print_helper = None
+
+    def set_verbosity(self, verbose=False, stat_func=None):
```
Comment (piiswrong) [inline python/mxnet/initializer.py]: input print func directly. no need for stat_func

File: python/mxnet/initializer.py [hunk 3]
```diff
@@ -79,17 +110,27 @@ def __call__(self, desc, arr):
...
             # when calling Variable initializer
             create(init)._init_weight(desc, arr)
+            if self._verbose:
+                logging.info('Initialized %s by %s: %s', desc, init, self._print_helper(arr))
         else:
             # register nnvm::FSetInputVariableAttrs in the backend for new patterns
             # don't add new cases here.
             if desc.endswith('weight'):
                 self._init_weight(desc, arr)
+                if self._verbose:
```
Comment (chunyang-wen) [inline python/mxnet/initializer.py]: Can we use python descriptor to implement this? It seems there are bunches of repeated codes.

Comment (zhreshold) [inline python/mxnet/initializer.py]: @chunyang-wen did you mean logging or init?

Comment (chunyang-wen) [inline python/mxnet/initializer.py]: Logging。

**Pull request merged**